pg_diffix/node_funcs.h

```diff
@@ -7,8 +7,12 @@
 /*
  * Returns `true` if the node represents a constant from pg_diffix perspective, i.e. `1` or `$1`.
  */
-extern bool is_simple_constant(Node *node);
+extern bool is_stable_expression(Node *node);
 
-extern void get_simple_constant_typed_value(Node *node, ParamListInfo bound_params, Oid *type, Datum *value, bool *isnull);
+/*
+ * Fills `type`, `value` and `isnull` with what the stable expression `node` holds.
+ * `bound_params` must be provided, since `node` might be a `Param` node.
+ */
+extern void get_stable_expression_typed_value(Node *node, ParamListInfo bound_params, Oid *type, Datum *value, bool *isnull);
```

Comment:
`get_stable_expression_value` should suffice. You can tell that you get the type from the out arg.

Reply:
:+1: took th eliberty to force push this one